### PR TITLE
Added Adler-32 benchmark.

### PR DESCRIPTION
**Pull request by 5hubh4m**

### CODE REVIEW

**Comment by sampsyo** [comment]
Super interesting! Nice work!

**Comment by 5hubh4m** [comment]
Writing this benchmark made me think we should have bitwise ops in the spec haha!

**Comment by sampsyo** [comment]
Indeed; the `@bitwise_or` function is a pretty good demonstration of that. 😂

File: benchmarks/adler32.bril [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+# Fills the array with values 0, 1. 2, ...
+@fill_array(arr: ptr<int>, size:int) {
+  one: int = const 1;
+  curr: int = const 0;
+  loc: ptr<int> = id arr;
+.loop:
+  store loc curr;
+  loc: prt<int> = ptradd loc one;
```
Comment (Pat-Lafon) [inline benchmarks/adler32.bril:25]: From brilirs: `prt` -> `ptr`

Comment (5hubh4m) [reply benchmarks/adler32.bril:25]: Interesting that `brili` didn't catch this.

Comment (Pat-Lafon) [reply benchmarks/adler32.bril:25]: My understanding is that `brili` does it's type checking based on the type of the value being evaluated and ignores the destination type.

Comment (sampsyo) [reply benchmarks/adler32.bril:25]: Very true; brili is purely dynamic. There are static type checkers elsewhere in the repository.

**Pull request merged**